libs/api/domains/rights-portal/src/lib/payment/dto/copaymentPeriod.input.ts

```diff
@@ -2,9 +2,9 @@ import { Field, InputType } from '@nestjs/graphql'
 
 @InputType('RightsPortalCopaymentPeriodInput')
 export class CopaymentPeriodInput {
-  @Field(() => String)
-  dateFrom!: string
+  @Field(() => Date)
+  dateFrom!: Date
 
-  @Field(() => String)
-  dateTo!: string
+  @Field(() => Date)
+  dateTo!: Date
 }
```

Comment:
Also this, `PaymentParticipation.tsx` formats the dates to string as well.

Reply:
I tested the frontend manually as thoroughly as I could and couldn't produce an error or a crash, however I might not have the most domain knowledge when it comes to the service portal. 

When testing PaymentParticipation.tsx the graphQL/NestJS layer seems to correctly parse the string formatted date to a Date object:
![image](https://github.com/island-is/island.is/assets/42970319/ed32c07a-e877-476e-8b42-c72509aa2abf)
![image](https://github.com/island-is/island.is/assets/42970319/7d518b2d-1d84-40c5-a437-3e81f71189ae)
![image](https://github.com/island-is/island.is/assets/42970319/ab81eafe-580d-47c7-883c-04e47f631fd4)

And returns what I believe to be the correct data:
![image](https://github.com/island-is/island.is/assets/42970319/c5ade7e6-d5ac-4674-9c57-df6d489317aa)

This test was performed using Gervimaður Færeyjar(010-2399)

I do find it a bit odd to format the dates to MM/dd/yyyy instead of something like ISO 8601 when making the request, is that something the rights-portal client requires?

Reply:
Not sure, but I would prefer if you could change the queries in `PaymentParticipation.tsx` and `PaymentOverview.tsx ` to use date objects instead of the formatted dates, much cleaner imo.